From 4946518453019aac7eb36cb2e16d8330c34247d7 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Wed, 15 Aug 2018 19:22:49 +0200
Subject: [PATCH 01/12] libbinder: Don't log call trace when waiting for vendor
 service on non-eng builds

* On some devices this results in a lot of unnecessary log spam.

Change-Id: Id797a2ce2590ebc718e3bf84c4edeb763ffd379d
---
 libs/binder/IServiceManager.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/libs/binder/IServiceManager.cpp b/libs/binder/IServiceManager.cpp
index be3bbf749..721841e44 100644
--- a/libs/binder/IServiceManager.cpp
+++ b/libs/binder/IServiceManager.cpp
@@ -157,12 +157,17 @@ public:
         // retry interval in millisecond.
         const long sleepTime = gSystemBootCompleted ? 1000 : 100;
 
+        char buildType[PROPERTY_VALUE_MAX];
+        property_get("ro.build.type", buildType, "unknown");
+        const bool isBuildTypeEng = strcmp(buildType, "eng") == 0;
+
         int n = 0;
         while (uptimeMillis() < timeout) {
             n++;
             if (isVendorService) {
+                property_get("ro.build.type", buildType, "unknown");
                 ALOGI("Waiting for vendor service %s...", String8(name).string());
-                CallStack stack(LOG_TAG);
+                if (isBuildTypeEng) CallStack stack(LOG_TAG);
             } else if (n%10 == 0) {
                 ALOGI("Waiting for service %s...", String8(name).string());
             }
-- 
2.17.1

